python/ray/autoscaler/_private/kuberay/node_provider.py

```diff
@@ -170,31 +208,152 @@ def __init__(
 
         self.headers, self.verify = load_k8s_secrets()
 
-        # Disallow multiple node providers, unless explicitly allowed for testing.
-        global provider_exists
-        if not _allow_multiple:
-            assert (
-                not provider_exists
-            ), "Only one KuberayNodeProvider allowed per process."
-        assert (
-            provider_config.get(DISABLE_NODE_UPDATERS_KEY, False) is True
-        ), f"To use KuberayNodeProvider, must set `{DISABLE_NODE_UPDATERS_KEY}:True`."
-        assert provider_config.get(DISABLE_LAUNCH_CONFIG_CHECK_KEY, False) is True, (
-            "To use KuberayNodeProvider, must set "
-            f"`{DISABLE_LAUNCH_CONFIG_CHECK_KEY}:True`."
-        )
-        assert (
-            provider_config.get(FOREGROUND_NODE_LAUNCH_KEY, False) is True
-        ), f"To use KuberayNodeProvider, must set `{FOREGROUND_NODE_LAUNCH_KEY}:True`."
         assert (
             provider_config.get(WORKER_LIVENESS_CHECK_KEY, True) is False
         ), f"To use KuberayNodeProvider, must set `{WORKER_LIVENESS_CHECK_KEY}:False`."
         assert (
             provider_config.get(WORKER_RPC_DRAIN_KEY, False) is True
         ), f"To use KuberayNodeProvider, must set `{WORKER_RPC_DRAIN_KEY}:True`."
-        provider_exists = True
+        BatchingNodeProvider.__init__(
```

Comment:
nit I think `__super__(...)` is the preferred way of doing this.

Reply:
I've seen both ways in the codebase, but I prefer to make it explicit.